accelerator/proxy_apis.c

```diff
@@ -322,50 +322,69 @@ static status_t api_get_trytes(const iota_client_service_t* const service, const
-  enum {
-    H_checkConsistency = 3512812181U,
-    H_findTransactions = 1314581663U,
-    H_getBalances = 185959358U,
-    H_getInclusionStates = 3627925933U,
-    H_getNodeInfo = 1097192119U,
-    H_getTrytes = 3016694096U
-  };
-
-  // With DJB2 hash function we could reduce the amount of string comparisons and deterministic execution time
-  switch (hash_algo_djb2(command)) {
-    case H_checkConsistency:
-      ret = api_check_consistency(service, obj, json_result);
-      break;
-    case H_findTransactions:
-      ret = api_find_transactions(service, obj, json_result);
-      break;
-    case H_getBalances:
-      ret = api_get_balances(service, obj, json_result);
-      break;
-    case H_getInclusionStates:
-      ret = api_get_inclusion_states(service, obj, json_result);
-      break;
-    case H_getNodeInfo:
-      ret = api_get_node_info(service, json_result);
-      break;
-    case H_getTrytes:
-      ret = api_get_trytes(service, obj, json_result);
-      break;
-
-    default:
-      ta_log_error("%s\n", "SC_HTTP_URL_NOT_MATCH");
-      return SC_HTTP_URL_NOT_MATCH;
+  if (iconf->proxy_api_direct_pass) {
+    char_buffer_t* res_buff = char_buffer_new();
+    char_buffer_t* req_buff = char_buffer_new();
+    char_buffer_set(req_buff, obj);
+
+    retcode_t result = iota_service_query(service, req_buff, res_buff);
+    if (result != RC_OK) {
+      ta_log_error("%s\n", error_2_string(result));
+      goto done;
+    }
+    char_buffer_to_str(res_buff, json_result);
+
+  done:
+    char_buffer_free(req_buff);
+    char_buffer_free(res_buff);
+  } else {
+    const uint8_t max_cmd_len = 30;
+    char command[max_cmd_len];
+
+    proxy_apis_command_req_deserialize(obj, command);
+    // TODO more proxy APIs should be implemented
+    enum {
+      H_checkConsistency = 3512812181U,
+      H_findTransactions = 1314581663U,
+      H_getBalances = 185959358U,
+      H_getInclusionStates = 3627925933U,
+      H_getNodeInfo = 1097192119U,
+      H_getTrytes = 3016694096U
+    };
+
+    // With DJB2 hash function we could reduce the amount of string comparisons and deterministic execution time
+    switch (hash_algo_djb2(command)) {
+      case H_checkConsistency:
+        ret = api_check_consistency(service, obj, json_result);
+        break;
+      case H_findTransactions:
+        ret = api_find_transactions(service, obj, json_result);
+        break;
+      case H_getBalances:
+        ret = api_get_balances(service, obj, json_result);
+        break;
+      case H_getInclusionStates:
+        ret = api_get_inclusion_states(service, obj, json_result);
+        break;
+      case H_getNodeInfo:
+        ret = api_get_node_info(service, json_result);
+        break;
+      case H_getTrytes:
+        ret = api_get_trytes(service, obj, json_result);
+        break;
+
+      default:
+        ta_log_error("%s\n", "SC_HTTP_URL_NOT_MATCH");
+        return SC_HTTP_URL_NOT_MATCH;
+    }
+
+    if (ret) {
+      ta_log_error("%d\n", ret);
```

Comment:
Add more messages to clarify the error here?

Reply:
what kinds of message should be added?

Reply:
Only one number is meanless.
Add the description of the error, which makes someone could know what happened.